src/app/features/students/students.component.ts

```diff
@@ -0,0 +1,71 @@
+import { Component, OnDestroy, OnInit, Signal, WritableSignal, computed, effect, inject, signal } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { FormsModule } from '@angular/forms';
+
+import { NzTypographyModule } from 'ng-zorro-antd/typography';
+import { NzTableModule } from 'ng-zorro-antd/table';
+import { NzDividerModule } from 'ng-zorro-antd/divider';
+import { NzButtonModule } from 'ng-zorro-antd/button';
+import { NzInputModule } from 'ng-zorro-antd/input';
+import { NzPopconfirmModule } from 'ng-zorro-antd/popconfirm';
+
+import { StudentService } from '../../shared/services/student.service';
+import { Student, StudentModalType } from '../../shared/interfaces/students';
+import { HeaderComponent } from '../../shared/components/header/header.component';
+import { NzMessageModule, NzMessageService } from 'ng-zorro-antd/message';
+import { StudentModalComponent } from './components/student-modal/student-modal.component';
+import { Subject, takeUntil } from 'rxjs';
+
+@Component({
+  selector: 'app-students',
+  standalone: true,
+  imports: [CommonModule, HeaderComponent, NzTypographyModule, NzTableModule, NzDividerModule, NzButtonModule, NzInputModule, FormsModule, CommonModule, NzMessageModule, NzPopconfirmModule, StudentModalComponent],
+  templateUrl: './students.component.html',
+  styleUrl: './students.component.sass'
+})
+export class StudentsComponent implements OnInit, OnDestroy {
+  private studentsService: StudentService = inject(StudentService)
+  private toast: NzMessageService = inject(NzMessageService)
+  
+  public isShowStudentModal: boolean = false;
+
+  public students: Signal<Student[]> = computed(() => this.studentsService.students())
+  public loading: Signal<boolean> = computed(() => this.studentsService.loading())
+  private error: Signal<string | null> = computed(() => this.studentsService.error())
+  public actionModal: WritableSignal<StudentModalType> = signal("add");
+
+  private onDestroy$: Subject<void> = new Subject();
+
+  constructor() {
+    effect(() => {
+      if (this.error()) {
+        this.toast.error(this.error() || '', {
+          nzDuration: 5000
+        })
+      }
+    })
+  }
+
+  public ngOnInit(): void {
+    this.studentsService.getStudents().pipe(takeUntil(this.onDestroy$)).subscribe();
+    this.studentsService.deleteStudent().pipe(takeUntil(this.onDestroy$)).subscribe();
```

Comment:
You don't need to subscribe on deleteStudent() on init. It's an action, you should subscribe on it only when you need to execute it. See, you've just initialized a component and you already have 3 subscriptions, 2 of which are odd.
No questions about the first one, but deleteStudent and selectStudent should be called on demand.